src/python/pants/backend/go/lint/gofmt/rules.py

```diff
@@ -86,8 +87,13 @@ async def gofmt_fmt(request: GofmtRequest, gofmt: GofmtSubsystem) -> FmtResult:
         return FmtResult.skip(formatter_name=request.name)
     setup = await Get(Setup, SetupRequest(request, check_only=False))
     result = await Get(ProcessResult, Process, setup.process)
-    return FmtResult.from_process_result(
-        result, original_digest=setup.original_digest, formatter_name=request.name
+    output_snapshot = await Get(Snapshot, Digest, result.output_digest)
+    return FmtResult(
+        setup.original_snapshot,
+        output_snapshot,
+        stdout=result.stdout.decode(),
+        stderr=result.stderr.decode(),
+        formatter_name=request.name,
```

Comment:
It would probably still be good to have the helper method take the result so that it can decode the stdout/stderr for you at the least (although obviously a bit redundant that the `Snapshot` needs to be precalculated... could validate that the argument `Snapshot` has a `Digest` equal to the result's `Digest`...?)

Reply:
The `stdout`/`stderr` isn't always decoded. Sometimes it is passed through `strip_v2_chroot`. See https://github.com/pantsbuild/pants/blob/b4ef1a1ebe557607ef376963f7093be25d682f75/src/python/pants/backend/java/lint/google_java_format/rules.py#L137